client-metrics/src/main/scala/org/http4s/client/metrics/Metrics.scala

```diff
@@ -0,0 +1,71 @@
+package org.http4s.client.metrics
+
+import cats.data.Kleisli
+import cats.effect._
+import cats.implicits._
+import com.codahale.metrics.{Counter, MetricRegistry, Timer}
+import java.util.concurrent.TimeUnit
+import org.http4s.{Request, Status}
+import org.http4s.client.{Client, DisposableResponse}
+
+object Metrics {
+  def apply[F[_]: Sync](registry: MetricRegistry, prefix: String = "org.http4s.client")(
+      client: Client[F]): Client[F] = {
+
+    def withMetrics(metrics: MetricsCollection)(req: Request[F]): F[DisposableResponse[F]] =
+      for {
+        start <- Sync[F].delay(System.nanoTime())
+        _ <- Sync[F].delay(metrics.activeRequests.inc())
+        resp <- client.open(req)
+        _ <- Sync[F].delay(metrics.activeRequests.dec())
```

Comment:
This will still be active through processing of the response body.  We'd want to hook into the dispose method.

Reply:
So, decreasing the active requests counter should be moved to the disponse handler of DisposableResponse right?

Reply:
So, decreasing the active requests counter should be moved to the disponse handler of DisposableResponse right?

Reply:
Yes.  Probably with `.guarantee`, just in case something throws.

Reply:
Could you elaborate on this please? What should we done with .guarantee? I mean updating the metrics happen before calling the original dispose method. So, as I see it, the only that go wrong here is updating the metrics itself and in that case the .guarantee would probably not help... Or what's your thinking here?

Reply:
The `guarantee` would help if the metrics call fails, because the guaranteed effect would run, and the original error would be re-raised.

I've been bitten a few times by Prometheus throwing, but it's always on the first call, and always because I did something stupid in naming.  So it's probably not worth worrying about.

Reply:
ok